upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -633,42 +577,46 @@ func (c *ApplyClusterCmd) Run() error {
 		var images []*nodeup.Image
 
 		if components.IsBaseURL(cluster.Spec.KubernetesVersion) {
-			baseURL := cluster.Spec.KubernetesVersion
-			baseURL = strings.TrimSuffix(baseURL, "/")
-
-			// TODO: pull kube-dns image
 			// When using a custom version, we want to preload the images over http
 			components := []string{"kube-proxy"}
 			if role == kops.InstanceGroupRoleMaster {
 				components = append(components, "kube-apiserver", "kube-controller-manager", "kube-scheduler")
 			}
+
 			for _, component := range components {
-				imagePath := baseURL + "/bin/linux/amd64/" + component + ".tar"
-				glog.Infof("Adding docker image: %s", imagePath)
+				baseURL, err := url.Parse(cluster.Spec.KubernetesVersion)
```

Comment:
Using URLs instead of catenating strings.  I wasted a lot of time debugging bad string catenations, or when I missed a period.  Now all of the URLs are handled as URLs and not strings.